src/Simulation/Native/src/external/fused.hpp

```diff
@@ -41,7 +41,14 @@ class Fused
       fusedgates = Fusion();
     }
 
+    Fusion get_fusedgates() const& {
+        return fusedgates;
+    }
     
+    void set_fusedgates(Fusion newFusedGates) const {
```

Comment:
declaring setters as `const` doesn't make sense: the purpose of the setter is to change the object, isn't it?

Reply:
This needs to be set as const, as I am calling it in flush() which is const. Yes, the setter is supposed to change the object - but in this case by change here we are changing the contents of the container, not the container or its location.

Reply:
As a part of _public class interface_ a const setter doesn't make much sense, whatever the implementation details of other methods might be. If clients outside of the class need to use the setter (for example, tests), make it non-const for them and inside `flush()` do the assignment directly, without going through the setter.

Reply:
Apologies, I did not really understand what you were trying to convey here completely. Unfortunately because the data structure fusedgates is private, I cannot make the assignments outside this class, thereby not allowing me to make assignments in flush() directly. Given this, the use of a setter is necessary to alter the datastructure. And as this setter is being called in a function which is declared const, the language does not allow me to have the setter to be non-const.

Reply:
"funny" public interfaces like this usually indicate that there is an issue with design of data structures: basically, you are trying to put a square peg into a round hole. You have multiple choices to address it:
1. keep entrenching the previous design decision(s) and add public interfaces that don't conform to good practices (ugly)
2. keep the previous design decision intact and workaround by adding const_cast(s) (ugly but obvious to spot in code and keeps badness from spreading)
3. remove `const` from member functions that violate constness (easy, as any chickening out is, but honest :))
4. rethink the logical\representational constness model of your data structures and get it right (hard and, given where C++ is re constness, the results might not be that different from option 3)

Personally, for a quick solution I'd go with either 2 or 3.
Your call.

Reply:
Thought about this a little. Given a couple of factors - what this branch is trying to achieve and considering that I have just a couple of weeks left, I think it will be good idea for me to create an issue outlining this very relevant issue you bring up which can be tackled later. The only reason I say that is because I feel this deviates a little from the original purpose of this branch, and that making this const change is going to take some time as it very deeply entrenched in the simulator codebase, and I am hoping to get a couple of more branches checked-in before the end of my internship, and want to leave enough time for that. Thank you!